lxd/endpoints/network_test.go

```diff
@@ -88,3 +90,23 @@ func newTCPListener(t *testing.T) *net.TCPListener {
 	require.NoError(t, err)
 	return listener
 }
+
+// Create IPv4 0.0.0.0 listener using random port
+// and ensure it is not accessible via IPv6 request.
+func TestEndpoints_NetworkCreateTCPSocketIPv4(t *testing.T) {
+	endpoints, config, cleanup := newEndpoints(t)
+	defer cleanup()
+
+	config.NetworkAddress = "0.0.0.0:0"
+	require.NoError(t, endpoints.Up(config))
+
+	address, certificate := endpoints.NetworkAddressAndCert()
+	parts := strings.Split(address, ":")
+	ipv6Address := fmt.Sprintf("[::1]:%s", parts[1])
+	message := fmt.Sprintf("Get https://%s/: dial tcp %s: connect: connection refused", ipv6Address, ipv6Address)
+
+	assert.NoError(t, httpGetOverTLSSocket(address, certificate))
```

Comment:
shouldn't this be 127.0.0.1 plus the port generated the same way you did for `ipv6Address` for symmetry?

Reply:
We can do that, but I don't see huge difference in doing this here, since we test for ipv4/ipv6. Anything you can see from your side?

Reply:
Just for consistency really, so the tests aren't reliant on the address returned from the helper.